app/src/main/java/com/nexenio/bleindoorpositioningdemo/ui/beaconview/BeaconViewFragment.java

```diff
@@ -147,8 +148,10 @@ protected void onColoringModeSelected(@ColorUtil.ColoringMode int coloringMode,
         for (Beacon beacon : beaconManager.getBeaconMap().values()) {
             for (BeaconFilter beaconFilter : beaconFilters) {
                 if (beaconFilter.matches(beacon)) {
-                    beacons.add(beacon);
-                    break;
+                    if (beacon.getMacAddress().equals("E2:38:2E:68:46:E9")) {
```

Comment:
This should be part of a `BeaconFilter` and only in the `BeaconChartFragment`. Preferably, it should select the closest beacon and not a hard coded MAC address.